src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractDatasetCommand.java

```diff
@@ -153,7 +156,7 @@ protected void validateOrDie(DatasetVersion dsv, Boolean lenient) throws Command
      * @param ctxt
      * @throws CommandException
      */
-    protected void registerExternalIdentifier(Dataset theDataset, CommandContext ctxt, boolean retry) throws CommandException {
+    protected void registerExternalIdentifier(DvObject theDataset, CommandContext ctxt, boolean retry) throws CommandException {
```

Comment:
FOr this change, the theDataset parameter name should probably change as well. That said, files don't currently (before #7334) get PIDs prior to publication, so file related code could be dropped for now.